### DISCUSSION TITLE
for looping an experiment with different images and trial duration

### DISCUSSION THREAD

**Discussion by Stajduhar-A** [jspsych/jsPsych#974] [Q&A]
Hi Josh,

I'm trying to become much more efficient when it comes to coding my experiments but I'm a bit stuck. Basically, what I have to do is present a fixation for 1000ms, then one image for 2000ms, then a mask for 500ms, then another image for 4000ms. I have defined my image arrays like so:  
<img width="607" alt="Screen Shot 2020-08-13 at 4 56 04 PM" src="https://user-images.githubusercontent.com/52038520/90186244-e1636d80-dd85-11ea-9a71-812443ed1f46.png">

I then defined my trial like so:
<img width="411" alt="Screen Shot 2020-08-13 at 4 56 49 PM" src="https://user-images.githubusercontent.com/52038520/90186333-fcce7880-dd85-11ea-8829-1be1ddaa0579.png">

I know this is wrong since it only takes into account one image array and loops through that without going onto the others. How do I alter this so that I get the order I mentioned above as well as the different trial durations associated for each image? Is the idea of having separate arrays to house everything the appropriate thing to do here, or is there a better way to go about it? Thanks for the help, Josh!

**Comment by jodeleeuw** [comment]
Hi @Stajduhar-A,

I think timeline variables will help you do what you are looking for here. In addition to the [documentation](https://www.jspsych.org/overview/timeline/#timeline-variables), they are also covered in [the tutorial](https://www.jspsych.org/tutorials/rt-task/#part-5-timeline-variables).

If you want video-based tutorials, then [day 1 of the moving research online workshop](https://www.youtube.com/watch?v=BuhfsIFRFe8) talks about it, as does my tutorial on the [dichotic listening task](https://www.youtube.com/watch?v=FH999P3QFx0&list=PLnfo1lBY1P2Mf_o6rV5wiqqn92Mw3UTGh&index=3&t=0s).

Happy to follow up on any questions you've got.
